src/services/admin/AuditLogsService.ts

```diff
@@ -0,0 +1,500 @@
+                siteName: repoName,
+                auditLogs,
+                snapshotTime: new Date(),
+              })
+            )
+          })
+        )
+      )
+      .andThen((auditLogDtos) => {
+        // Step 5: Prepare the audit log CSV files for each repo
+        const auditLogHeader = "Date,Time,Activity,User,Page,Remarks\n"
+
+        return ResultAsync.combine(
+          auditLogDtos.map(({ siteName, auditLogs, snapshotTime }) => {
+            const csvContent = auditLogs.reduce(
+              (csv, { timestamp, activity, actor, page, remarks }) => {
+                const recordDate = timestamp.toISOString().split("T")[0]
+                const recordHour = timestamp
+                  .getUTCHours()
+                  .toString()
+                  .padStart(2, "0")
+                const recordMinute = timestamp
+                  .getUTCMinutes()
+                  .toString()
+                  .padStart(2, "0")
+                const recordSecond = timestamp
+                  .getUTCSeconds()
+                  .toString()
+                  .padStart(2, "0")
+                const recordTime = `${recordHour}:${recordMinute}:${recordSecond} (UTC)`
+
+                return `${csv}${recordDate},${recordTime},${activity},"${actor}","${page}","${remarks}"\n`
+              },
+              auditLogHeader
+            )
+            const csvFileDate = snapshotTime.toISOString().split("T")[0]
+            const csvFileName = `audit-logs_${siteName}_${csvFileDate}_${formSubmissionId}.csv`
+            const csvFilePath = path.join(EFS_VOL_PATH_AUDIT_LOGS, csvFileName)
+
+            return ResultAsync.fromPromise(
+              fs.promises.writeFile(csvFilePath, csvContent, "utf-8"),
+              (error) => {
+                logger.error(
+                  `Error occurred while writing audit log CSV file for repo ${siteName}: ${JSON.stringify(
+                    error
+                  )}`
+                )
+                return new AuditLogsError(
+                  `Error occurred while writing audit log CSV file for repo ${siteName}`
+                )
+              }
+            ).map(() => csvFilePath)
+          })
+        )
+      })
+      .andThen((csvFilePaths) => {
+        // Step 6: Send the audit log CSV files to the user via email
+        const emailBody = `<p>Please find the attached audit log CSV files for the requested repos:</p>
+        <ul>
+          ${repoNames.map((repoName) => `<li>${repoName}</li>`)}
+        </ul>
+        <p>Isomer Support Team</p>`
+
+        return ResultAsync.fromPromise(
+          mailer.sendMail(email, emailSubject, emailBody, csvFilePaths),
+          () =>
+            new AuditLogsError(
+              `Error occurred while sending audit log CSV files to ${email} (submission ID: ${formSubmissionId}). Requested repos: ${repoNames.join(
+                ", "
+              )}`
+            )
+        )
+      })
+      .orElse((error) => {
```

Comment:
error here is typed to a `unknown`, i would have expected a obj/string... thoughts?

Reply:
also we are using result.combine which short circuits! this means that we will either send the success state if all the repos pass, or fail for when only one fails. this means that repos that pass, will sill be reported as failed here due to this control flow. is this intended?

Reply:
Not sure why the typings died halfway through, adjusted in 6a46e291197690df3fc3825f46e0efc311bab2be.

The short circuiting is intended! I'm erring on the side of caution here to fail often and only provide the logs if everything is successful.

Reply:
sure
noticed that we are returning the error directly to the user. have we ensured that we are not returning cryptic/sensitive error info? eg. some "sequalize error string expected" shouldnt reach our user
also, have we set up monitoring for this feature? will us devs be aware if suddenly all of our forms are failing?

Reply:
Decided to add a common prefix for error logs here so we can just monitor for this prefix to see if any errors occur: 24658ec51af72594c7da2c6368ebc0f7be5d5c52.

We are not returning sensitive/cryptic information back to the user, because the error that is returned only contains text that we define and not the actual error message that comes from say sequelize.